src/sentry/api/endpoints/organization_health.py

```diff
@@ -231,18 +231,30 @@ def get_environments(self, request, organization):
         if not requested_environments:
             return []
 
-        environments = dict(
+        environments = set(
             Environment.objects.filter(
                 organization_id=organization.id,
                 name__in=requested_environments,
-            ).values_list('name', 'id'),
+            ).values_list('name', flat=True),
         )
 
-        if requested_environments != set(environments.keys()):
+        if requested_environments != environments:
             raise ResourceDoesNotExist
 
-        # snuba requires ids for filter keys
-        return environments.values()
+        conditions = []
+
+        if '' in environments:
+            environments.remove('')
+            conditions.append(['tags[environment]', 'IS NULL', None])
+
+        if environments:
+            conditions.append(['tags[environment]', 'IN', list(environments)])
+
+        # OR clauses are nested
```

Comment:
Can you not always `return [conditions]` even if there's 1 item? It seems a little more intuitive to me, instead of only doing it when there are 2. I assume an OR with 1 item won't break and won't change the result, since that doesn't really make sense.